src/utils/api/transactions/TransactionResultUtil.ts

```diff
@@ -74,46 +76,73 @@ export const getOnboardingPaymentOutcome = (
-    case TransactionStatusEnum.REFUNDED:
+        : ViewOutcomeEnum.PSP_ERROR;
     case TransactionStatusEnum.REFUND_REQUESTED:
     case TransactionStatusEnum.REFUND_ERROR:
       return ViewOutcomeEnum.GENERIC_ERROR;
+    case TransactionStatusEnum.REFUNDED:
+    case TransactionStatusEnum.NOTIFIED_KO:
+      return ViewOutcomeEnum.PSP_ERROR;
     case TransactionStatusEnum.EXPIRED_NOT_AUTHORIZED:
       return ViewOutcomeEnum.TIMEOUT;
     case TransactionStatusEnum.CANCELED:
     case TransactionStatusEnum.CANCELLATION_EXPIRED:
       return ViewOutcomeEnum.CANCELED_BY_USER;
-    case TransactionStatusEnum.CLOSURE_ERROR:
     case TransactionStatusEnum.CLOSURE_REQUESTED:
+      return !wasAuthorizedByGateway(gateway, gatewayAuthorizationStatus)
+        ? evaluateUnauthorizedStatus(
+            gateway,
+            errorCode,
+            gatewayAuthorizationStatus
+          )
+        : ViewOutcomeEnum.TAKING_CHARGE;
+    case TransactionStatusEnum.CLOSURE_ERROR:
     case TransactionStatusEnum.AUTHORIZATION_COMPLETED:
-    case TransactionStatusEnum.UNAUTHORIZED:
       return !wasAuthorizedByGateway(gateway, gatewayAuthorizationStatus)
         ? evaluateUnauthorizedStatus(
             gateway,
             errorCode,
             gatewayAuthorizationStatus
           )
         : ViewOutcomeEnum.GENERIC_ERROR;
+    case TransactionStatusEnum.UNAUTHORIZED:
+      return !wasAuthorizedByGateway(gateway, gatewayAuthorizationStatus)
+        ? evaluateUnauthorizedStatus(
+            gateway,
+            errorCode,
+            gatewayAuthorizationStatus
+          )
+        : ViewOutcomeEnum.PSP_ERROR;
     case TransactionStatusEnum.CLOSED:
       return sendPaymentResultOutcome ===
         SendPaymentResultOutcomeEnum.NOT_RECEIVED
         ? ViewOutcomeEnum.TAKING_CHARGE
         : ViewOutcomeEnum.GENERIC_ERROR;
     case TransactionStatusEnum.EXPIRED: {
-      if (!wasAuthorizedByGateway(gateway, gatewayAuthorizationStatus)) {
+      if (gatewayAuthorizationStatus == null) {
+        return ViewOutcomeEnum.TAKING_CHARGE;
+      } else if (!wasAuthorizedByGateway(gateway, gatewayAuthorizationStatus)) {
         return evaluateUnauthorizedStatus(
           gateway,
           errorCode,
           gatewayAuthorizationStatus
         );
-      }
-      if (
+      } else if (
         sendPaymentResultOutcome === SendPaymentResultOutcomeEnum.OK &&
         wasAuthorizedByGateway(gateway, gatewayAuthorizationStatus)
       ) {
         return ViewOutcomeEnum.SUCCESS;
+      } else if (
+        sendPaymentResultOutcome === SendPaymentResultOutcomeEnum.KO &&
+        wasAuthorizedByGateway(gateway, gatewayAuthorizationStatus)
+      ) {
+        return ViewOutcomeEnum.PSP_ERROR;
+      } else if (
+        sendPaymentResultOutcome ===
+          SendPaymentResultOutcomeEnum.NOT_RECEIVED &&
+        wasAuthorizedByGateway(gateway, gatewayAuthorizationStatus)
+      ) {
+        return ViewOutcomeEnum.TAKING_CHARGE;
       }
```

Comment:
How about using a switch case?

```suggestion
     } else {
          switch (sendPaymentResultOutcome) {
            case SendPaymentResultOutcomeEnum.OK:
              return ViewOutcomeEnum.SUCCESS;
            case SendPaymentResultOutcomeEnum.KO:
              return ViewOutcomeEnum.PSP_ERROR;
            case SendPaymentResultOutcomeEnum.NOT_RECEIVED:
              return ViewOutcomeEnum.TAKING_CHARGE;
            default:
              return ViewOutcomeEnum.GENERIC_ERROR; // BE_KO(99)
          }
     }
```